### PR TITLE
feat(twap): validation and warnings - sell amount too small

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Part of #2628 

Implementing warning for [sell amount part too small](https://www.notion.so/cownation/FE-TWAP-Orders-c9f6bd5c23bd4c749147611dcb63daac?pvs=4#89819033ad944da2a82d4eb6aef78994) 

- Block action with text
- Show warning banner
Screenshot 2023-06-14 at 14 12 31

# To Test

1. Load as Safe app on Goerli (and hope USD quote is working 🤞)
2. Pick as sell amount a small enough amount OR pick a high enough number of parts until it's less than 5k
* Banner should show
* Action button text should be updated and disabled

## Notes
- The USDC calculation is not working all of the time on Goerli. Yesterday for example I had values fine, but today it's not showing up (getting 400 - low liquidity from quote endpoint)
- Banner text is proposed by me, suggestions are welcome, not closed on this

### CODE REVIEW

File: src/modules/twap/containers/TwapFormWidget/index.tsx [hunk 1]
```diff
@@ -109,6 +113,11 @@ export function TwapFormWidget() {
         </TradeTextBox>
       </styledEl.DeadlineRow>
 
+      <div>
```
Comment (shoom3301) [inline src/modules/twap/containers/TwapFormWidget/index.tsx:116]: Would be better to move it to a separate container: `<TwapFormWarnings />`

File: src/modules/twap/hooks/useTwapFormState.ts [hunk 2]
```diff
@@ -3,26 +3,37 @@ import { useMemo } from 'react'
...
   const twapOrder = useAtomValue(twapOrderAtom)
+  const { inputCurrencyFiatAmount } = useAdvancedOrdersDerivedState()
 
   const verification = useAsyncMemo(
     () => (extensibleFallbackContext ? verifyExtensibleFallback(extensibleFallbackContext) : null),
     [extensibleFallbackContext],
     null
   )
 
+  const sellAmountPartFiat = useMemo(() => {
```
Comment (shoom3301) [inline src/modules/twap/hooks/useTwapFormState.ts:29]: We already have it in `partsStateAtom`

File: src/modules/twap/containers/TwapFormWidget/index.tsx (hunk 1 above)
Comment (shoom3301) [reply src/modules/twap/containers/TwapFormWidget/index.tsx:116]: I also would like to display banners in this component, for example:
image

p.s. I'm working on it right now

Comment (shoom3301) [reply src/modules/twap/containers/TwapFormWidget/index.tsx:116]: Also, `getTwapFormState` should be renamed to `getTwapFormValidation` and `TwapFormState` => `TwapFormValidation`. But let's do it after merging our PR's :) Just to reduce friction

Comment (alfetopito) [reply src/modules/twap/containers/TwapFormWidget/index.tsx:116]: Hmmm ok.
I'll wait for your PR to update this one then

File: src/modules/twap/pure/PrimaryActionButton/getTwapFormState.tsx [hunk 3]
```diff
@@ -28,5 +37,9 @@ export function getTwapFormState(props: TwapFormStateParams): TwapFormState | nu
     return TwapFormState.NEED_FALLBACK_HANDLER
   }
 
+  if (isSellAmountTooSmall(sellAmountPartFiat, chainId)) {
```
Comment (shoom3301) [inline src/modules/twap/pure/PrimaryActionButton/getTwapFormState.tsx:40]: Also, loading of fiat prices takes some time. I think we shouldn't allow placing an order while a fiat price is not loading. Maybe we should return `TwapFormState.LOADING_SAFE_INFO`. It's not a blocker now, you can just leave a TODO

**Review by shoom3301 — APPROVED** [review]

File: src/modules/twap/pure/PrimaryActionButton/getTwapFormState.tsx (hunk 3 above)
Comment (alfetopito) [reply src/modules/twap/pure/PrimaryActionButton/getTwapFormState.tsx:40]: I don't think that should be a blocker because it might be down.
At least this is quite common with Goerli.
Yes, testnet, but we for instance don't block trades on mainnet when USD amounts are missing

**Comment by elena-zh** [comment]
Hey @alfetopito , just to clarify: for Gnosis chain min chunk should be 5$. Will it be implemented a bit later?

Also, can it be reverted to a lower amount for Goerly (for testing purposes)? :)

**Comment by elena-zh** [comment]
Also, I can't get the banner right now. But based on the screenshot you provided, could you please add a space between the text and an amount? 
image

One more edge case that comes to my mind: if we are on the Review modal, and a price is updated (or USD estimation), and a part becomes less than 5k, should we disable the trade or let a user place it anyways? 

Thanks!

**Comment by elena-zh** [comment]
> Oh, I completely misread that. No idea where I took this value from o.O Thanks for noticing, will update
Just in case, here is the spec :) 
image

**Comment by alfetopito** [comment]
> Also, I can't get the banner right now. But based on the screenshot you provided, could you please add a space between the text and an amount? image

True, I fixed that, just forgot to update the banner
 
> One more edge case that comes to my mind: if we are on the Review modal, and a price is updated (or USD estimation), and a part becomes less than 5k, should we disable the trade or let a user place it anyways?

I have the same question and I don't know the answer.
Can you raise this in the twap channel?

File: src/modules/twap/const.ts [hunk 4]
```diff
@@ -33,3 +35,9 @@ export const TWAP_HANDLER_ADDRESS: Record<SupportedChainId, string> = {
 }
 
 export const TWAP_PENDING_STATUSES = [TwapOrderStatus.WaitSigning, TwapOrderStatus.Pending, TwapOrderStatus.Scheduled]
+
+export const MINIMUM_PART_SELL_AMOUNT_FIAT: Record<SupportedChainId, CurrencyAmount<Currency>> = {
+  [SupportedChainId.MAINNET]: CurrencyAmount.fromRawAmount(USDC[SupportedChainId.MAINNET], 5_000e6), // 5k
+  [SupportedChainId.GOERLI]: CurrencyAmount.fromRawAmount(USDC[SupportedChainId.GOERLI], 100e6), // 100
+  [SupportedChainId.GNOSIS_CHAIN]: CurrencyAmount.fromRawAmount(USDC[SupportedChainId.GNOSIS_CHAIN], 5e6), // 5
```
Comment (alfetopito) [inline src/modules/twap/const.ts:42]: Using right values this time for goerli and gchain

**Comment by alfetopito** [comment]
@elena-zh updated

Prices aren't currently loading for me so I can't test, I assume it'll be the same for you 🤷

**Review by shoom3301 — APPROVED** [review]

**Comment by elena-zh** [comment]
Hey @alfetopito 
> > One more edge case that comes to my mind: if we are on the Review modal, and a price is updated (or USD estimation), and a part becomes less than 5k, should we disable the trade or let a user place it anyways?
> 
> I have the same question and I don't know the answer. Can you raise this in the twap channel?

I have created an issue for this #2772
